PROTOCOL.md

```diff
@@ -416,25 +426,25 @@ When the table property `delta.appendOnly` is set to `true`:
   - New log entries may rearrange data (i.e. `add` and `remove` actions where `dataChange=false`).
 
 ## Column Invariants
- - The schema for a given column MAY the metadata `delta.invariants`.
- - This column SHOULD be parsed as a boolean SQL expression.
+ - The `metadata` for a given column in the table schema MAY contain the key `delta.invariants`.
+ - This value of `delta.invariants` SHOULD be parsed as a boolean SQL expression.
  - Writers MUST abort any transaction that adds a row to the table, where a present invariant evaluates to `false` or `null`.
 
+## Generated Columns
+
+ - The `metadata` for a given column in the table schema MAY contain the key `delta.generationExpression`.
+ - This value of `delta.generationExpression` SHOULD be parsed as a boolean SQL expression.
+ - Writers MUST enforce any data writing to the table satisfy the condition `(<value> <=> <generation expression>) IS TRUE`.
+
 ## Writer Version Requirements
 
 The requirements of the writers according to the protocol versions are summarized in the table below. Each row inherits the requirements from the preceding row.
 
-+------------------+----------------------------------------------+
-|                  | Reader Version 1                             |
-+------------------+----------------------------------------------+
-| Writer Version 2 |  - Support `delta.appendOnly`                |
-|                  |  - Support column invariants                 |
-+------------------+----------------------------------------------+
-| Writer Version 3 |  - Enforce:                                  |
-|                  |    - `delta.checkpoint.writeStatsAsJson`     |
-|                  |    - `delta.checkpoint.writeStatsAsStruct`   |
-|                  |    - `CHECK` constraints                     |
-+------------------+----------------------------------------------+
+<br> | Reader Version 1
+-|-
+Writer Version 2 | - Support [`delta.appendOnly`](#append-only-tables)<br>- Support [Column Invariants](#column-invariants)
+Writer Version 3 | Enforce:<br>- `delta.checkpoint.writeStatsAsJson`<br>- `delta.checkpoint.writeStatsAsStruct`<br>- `CHECK` constraints
```

Comment:
`<br>` is to start a new line in a table cell.

Reply:
I've been searching for this html / markdown trick for new lines in tables for so long. Thank you @zsxwing !